### PR TITLE
Support for multiple slides per page

### PR DESCRIPTION
**Pull request by tamar-tchelidze**

### CODE REVIEW

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/SliderWidget/SliderWidget.component.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4790432&bid=15950279)

File: src/app/component/Slider/Slider.component.js [hunk 1]
```diff
@@ -21,6 +21,12 @@ import './Slider.style';
 export const ANIMATION_DURATION = 300;
 export const ACTIVE_SLIDE_PERCENT = 0.1;
 
+import {
+    TABLET_WIDTH,
+    DESKTOP_WIDTH
+} from 'Component/SliderWidget/SliderWidget.component';
```
Comment (Tomburgs) [inline src/app/component/Slider/Slider.component.js]: Imports come before exports.

File: src/app/component/Slider/Slider.component.js [hunk 2]
```diff
@@ -31,7 +37,10 @@ export default class Slider extends PureComponent {
         activeImage: PropTypes.number,
         onActiveImageChange: PropTypes.func,
         mix: MixType,
-        children: ChildrenType.isRequired
+        children: ChildrenType.isRequired,
+        slidesOnDesktop: PropTypes.number,
+        slidesOnTablet: PropTypes.number,
+        slidesOnMobile: PropTypes.number
```
Comment (Tomburgs) [inline src/app/component/Slider/Slider.component.js]: Props which are not explicitly set as `isRequired` need default props.

File: src/app/component/Slider/Slider.component.js [hunk 3]
```diff
@@ -116,6 +137,26 @@ export default class Slider extends PureComponent {
         }
     }
 
+    updateWindowDimensions = () => {
+        const { slidesOnDesktop, slidesOnTablet, slidesOnMobile } = this.props;
+        const { offsetWidth: sliderWidth = 0 } = this.draggableRef.current || {};
+        this.sliderWidth = sliderWidth;
```
Comment (Tomburgs) [inline src/app/component/Slider/Slider.component.js]: New line.

File: src/app/component/Slider/Slider.component.js [hunk 4]
```diff
@@ -116,6 +137,26 @@ export default class Slider extends PureComponent {
...
+        } else if (window.innerWidth >= TABLET_WIDTH) {
+            const slidesQtyPerPage = slidesOnTablet || 1;
+            this.setState({ slidesQtyPerPage });
+            this.slideWidth = sliderWidth / slidesQtyPerPage;
+        } else {
+            const slidesQtyPerPage = slidesOnMobile || 1;
+            this.setState({ slidesQtyPerPage });
+            this.slideWidth = sliderWidth / slidesQtyPerPage;
+        }
+        this.changeActiveImage(0);
```
Comment (Tomburgs) [inline src/app/component/Slider/Slider.component.js]: New line.

File: src/app/component/Slider/Slider.component.js [hunk 5]
```diff
@@ -277,6 +327,8 @@ export default class Slider extends PureComponent {
             showCrumbs, mix, activeImage, children
         } = this.props;
 
+        this.slidesQty = this.props.children.length;
```
Comment (Tomburgs) [inline src/app/component/Slider/Slider.component.js]: There is really no need to do this on each render, put it under `componentDidUpdate`.
Also - `this.slidesQty = children.length`?

File: src/app/component/SliderWidget/SliderWidget.component.js [hunk 6]
```diff
@@ -58,6 +105,7 @@ export default class SliderWidget extends PureComponent {
               block="SliderWidget"
               elem="Figure"
               key={ i }
+              style={ { width: `${this.state.slideWidth}%` } }
```
Comment (Tomburgs) [inline src/app/component/SliderWidget/SliderWidget.component.js]: Setting style this way will prevent Slider component in itself from being used to display multiple simultaneous slides easily.
I suggest instead create a ref map with children in Slider component and calculate the slide width in px there and set it that way.

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/SliderWidget/SliderWidget.component.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4790432&bid=15950279)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/SliderWidget/SliderWidget.component.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4790432&bid=15950279)

File: src/app/component/Slider/Slider.component.js (hunk 1 above)
Comment (alfredsgenkins) [reply src/app/component/Slider/Slider.component.js]: That is true, while you should still prefer putting path imports after aliased imports:

```js
import {
    TABLET_WIDTH,
    DESKTOP_WIDTH
} from 'Component/SliderWidget/SliderWidget.component';

import './Slider.style';
```

File: src/app/component/SliderWidget/SliderWidget.component.js (hunk 6 above)
Comment (alfredsgenkins) [reply src/app/component/SliderWidget/SliderWidget.component.js]: That is correct + I would recommend using CSS variables, not direct property setting.

**Comment by alfredsgenkins** [comment]
Closing for inactivity

**Comment by alfredsgenkins** [comment]
This PR is related: https://github.com/scandipwa/slider-graphql/pull/1

**Comment by alfredsgenkins** [comment]
Closing, QA failed, latest fixes with conflicts resolved available on `multiple-slides` branch.

**Pull request closed without merging**